### PR TITLE
8272439: G1: add documentation to G1CardSetInlinePtr

### PR DESCRIPTION
**Pull request by albertnetymk**
Add some documentation to `class G1CardSetInlinePtr`, and simplify a field initialization in `G1CardSetConfiguration`.

Test: hotspot_gc

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8272439](https://bugs.openjdk.java.net/browse/JDK-8272439): G1: add documentation to G1CardSetInlinePtr

### Reviewers
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**) ⚠️ Review applies to e282d72d1386ef01ef5602d58abb8f1361b2c289
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**)

### Contributors
 * Thomas Schatzl `<tschatzl@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5108/head:pull/5108` \
`$ git checkout pull/5108`

Update a local copy of the PR: \
`$ git checkout pull/5108` \
`$ git pull https://git.openjdk.java.net/jdk pull/5108/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5108`

View PR using the GUI difftool: \
`$ git pr show -t 5108`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5108.diff">https://git.openjdk.java.net/jdk/pull/5108.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/g1/g1CardSet.hpp [hunk 1]
```diff
@@ -47,6 +47,7 @@ enum G1AddCardResult {
 };
 
 class G1CardSetConfiguration {
+  // #bits required to hold the max card index; log(#cards_per_region)
```
Comment (tschatzl) [inline src/hotspot/share/gc/g1/g1CardSet.hpp:50]: I would prefer a real sentence like `// [Holds] the number of bits required to cover the maximum card index for the regions covered by this card set.`.

File: src/hotspot/share/gc/g1/g1CardSetContainers.hpp [hunk 2]
```diff
@@ -36,6 +36,38 @@
...
+// -> 14 bits per card; must have enough bits to hold the max card index
+// -> may encode up to 4 cards into it, using 61 bits (5 bits header + 4 * 14)
+//
+// M                                                     L
+// S                                                     S
+// B                                                     B
+// +------+         +---------------+--------------+-----+
+// |unused|   ...   |  card_index1  | card_index0  |SSS00|
+// +------+         +---------------+--------------+-----+
+
```
Comment (tschatzl) [inline src/hotspot/share/gc/g1/g1CardSetContainers.hpp:70]: Unnecessary extra newline

**Review by tschatzl — APPROVED** [review]
Some minor comments, but lgtm otherwise.

File: src/hotspot/share/gc/g1/g1CardSet.hpp (hunk 1 above)
Comment (albertnetymk) [reply src/hotspot/share/gc/g1/g1CardSet.hpp]: Fixed.

File: src/hotspot/share/gc/g1/g1CardSetContainers.hpp (hunk 2 above)
Comment (albertnetymk) [reply src/hotspot/share/gc/g1/g1CardSetContainers.hpp]: Removed.

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**